src/grpc.rs

```diff
@@ -122,6 +146,27 @@ where
     deserializer.deserialize_any(DurationVisitor)
 }
 
+fn serialize_optional_duration<S>(
+    value: &Option<Duration>,
+    serializer: S,
+) -> Result<S::Ok, S::Error>
+where
+    S: Serializer,
+{
+    if let Some(value) = value.as_ref() {
+        serialize_duration(value, serializer)
+    } else {
+        serializer.serialize_none()
+    }
+}
+
+fn deserialize_optional_duration<'de, D>(deserializer: D) -> Result<Option<Duration>, D::Error>
```

Comment:
**QUESTION:** Probably dumb question, but in which case we'll be deserialising this value? As I understood, we should be only sending it. Is it Rust convention to provide always both serialisation and deserialisation?

Reply:
We only do this when testing. There are fixtures/expectations we deserialize.

Reply:
Hm 🤔 Could you point me to those tests?

Reply:
https://github.com/EventStore/EventStoreDB-Client-Rust/blob/deadlines/src/grpc.rs#L43